src/pymor/models/iosys.py

```diff
@@ -540,21 +581,42 @@ def poles(self, mu=None):
         if not isinstance(mu, Mu):
             mu = self.parameters.parse(mu)
         assert self.parameters.assert_compatible(mu)
-        A = self.A.assemble(mu=mu)
-        E = self.E.assemble(mu=mu)
-
-        if self.order >= sparse_min_size():
-            if not isinstance(A, NumpyMatrixOperator) or A.sparse:
-                self.logger.warning('Converting operator A to a NumPy array.')
-            if not isinstance(E, IdentityOperator):
-                if not isinstance(E, NumpyMatrixOperator) or E.sparse:
-                    self.logger.warning('Converting operator E to a NumPy array.')
+        poles = self.presets['poles'] if 'poles' in self.presets else self._poles(mu=mu)
+        assert isinstance(poles, np.ndarray) and poles.shape == (self.A.source.dim,)
 
-        A = to_matrix(A, format='dense')
-        E = None if isinstance(E, IdentityOperator) else to_matrix(E, format='dense')
-        return spla.eigvals(A, E)
+        return poles
 
     @cached
+    def _gramian(self, typ, mu=None):
+        if typ == 'c_lrcf' and 'c_dense' in self.presets:
+            return self.A.source.from_numpy(_chol(self.presets['c_dense']).T)
+        elif typ == 'o_lrcf' and 'o_dense' in self.presets:
+            return self.A.source.from_numpy(_chol(self.presets['o_dense']).T)
+        elif typ == 'c_dense' and 'c_lrcf' in self.presets:
+            return self.presets['c_lrcf'].to_numpy().T @ self.presets['c_lrcf']
+        elif typ == 'o_dense' and 'o_lrcf' in self.presets:
+            return self.presets['o_lrcf'].to_numpy().T @ self.presets['o_lrcf']
```

Comment:
```suggestion
        elif typ == 'c_dense' and 'c_lrcf' in self.presets:
            return self.presets['c_lrcf'].to_numpy().T @ self.presets['c_lrcf'].to_numpy()
        elif typ == 'o_dense' and 'o_lrcf' in self.presets:
            return self.presets['o_lrcf'].to_numpy().T @ self.presets['o_lrcf'].to_numpy()
```

Reply:
I guess these branches weren't covered with tests before? Would that be easy to add?